rom/dev/tests/test_update_reset.rs

```diff
@@ -0,0 +1,211 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_builder::{FwId, ImageOptions, APP_WITH_UART, ROM_WITH_UART};
+use caliptra_error::CaliptraError;
+use caliptra_hw_model::{BootParams, Fuses, HwModel, InitParams, SecurityState};
+pub mod helpers;
+
+#[test]
+fn test_update_reset_success() {
+    pub const TEST_FMC_WITH_UART: FwId = FwId {
+        crate_name: "caliptra-rom-test-fmc",
+        bin_name: "caliptra-rom-test-fmc",
+        features: &["emu"],
+        workspace_dir: None,
+    };
+
+    let fuses = Fuses::default();
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART).unwrap();
+    let mut hw = caliptra_hw_model::new(BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            security_state: SecurityState::from(fuses.life_cycle as u32),
+            ..Default::default()
+        },
+        fuses,
+        ..Default::default()
+    })
+    .unwrap();
+
+    let image_bundle = caliptra_builder::build_and_sign_image(
+        &TEST_FMC_WITH_UART,
+        &APP_WITH_UART,
+        ImageOptions::default(),
+    )
+    .unwrap();
+
+    assert!(hw
+        .upload_firmware(&image_bundle.to_bytes().unwrap())
+        .is_ok());
+
+    hw.step_until_output_contains("[exit] Launching FMC")
+        .unwrap();
+
+    let result = hw.mailbox_execute(0x1000_0004, &[]);
```

Comment:
Consider defining this constant somewhere:

```
const TEST_FMC_CMD_RESET_FOR_UPDATE: u32 = 0x1000_0004 
```

Reply:
Updated; in a later PR, I'll maintain these constants in a common file which can referenced by both tests and test-fmc.